runner/src/main.rs

```diff
@@ -289,10 +303,28 @@ fn run_example_server(application_file: &str) -> Cmd {
     )
 }
 
+#[derive(serde::Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
 struct Example {
     name: String,
-    rust_module_names: Vec<String>,
-    additional_client_flags: Vec<String>,
+    modules: HashMap<String, Target>,
+    clients: HashMap<String, Client>,
+}
+
+#[derive(serde::Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+enum Target {
+    Bazel { bazel_target: String },
+    Cargo { cargo_manifest: String },
+}
```

Comment:
What I like about the manifest is that it specifies everything needed to run an example in one place, as opposed if-conditions in our various build scripts. :) 

With that approach, it has occurred to me that maybe it would make sense to keep the target logic outside of the runner and instead in each manifest. 

Eg instead of specifying a target, and the runner having instructions on how to build with that target, could the manifest simply list a command that should be run, leaving the details of the build system up to it?

Lemme know if that makes sense. :) 

Also curious to hear @rbehjati thoughts, since this somewhat relates to our conversation in the Oak Channel.

Reply:
Sorry for the delay. I like what you are proposing @juliettepretot. Whenever there is a discussion of reuse, it reminds me of this [hilarious talk](https://vimeo.com/233817459). 
I am also a bit worried about lines like this (although I think this line can easily be removed, and be added to each manifest):
https://github.com/project-oak/oak/blob/2709157404874ad8a7f34f9b6c6f102f9d78f096/runner/src/main.rs#L408

Reply:
Thanks for sharing, that was a fun talk. :) 

Re the line you mentioned, that's unrelated to the manifest. The same line exists in our bash script, and thankfully it will be phased out as we switch away from bazel.

Reply:
I think `--ca_cert=...` could be added to the `additional_args` list for each example, to make the manifests self-contained. 

It is not an issue now, especially since we are going to remove bazel. But I think in general it is better to keep the manifests self-contained. I am also thinking of these manifest files as a form of documentation and a reference for how to run the examples. Similar to what you wrote :) 

> Then in order to contribute an example, one needs to merely understand the manifest, not have a mental model of all of our build scripts.